### ISSUE TITLE
[iOS] Ad load failed with errorcode 1 when ATTracking status is denied.

### ISSUE THREAD

**Issue by Dooya-log** [googleads/googleads-mobile-unity#2922]
### Step 0: Are you in the right place?

  * For general technical questions, or help with project-specific issues like setting up ads in
   your app, reach out to our support team on the
   [Developer Forum](https://groups.google.com/forum/#!categories/google-admob-ads-sdk/game-engines).
  * For assistance with your AdMob account, reach out to
   [AdMob Support](https://support.google.com/admob/?hl=en#topic=7383088).
  * For feedback on [our documentation](https://developers.google.com/admob/unity/start),
   send  your feedback by pressing the **Send Feedback** button at the top right of the
    documentation page you are on.
  * For issues related to __the code in this repository__, continue filing this GitHub issue.

  * Once you've read this section and determined that your issue is appropriate for
    this repository, **please delete this section**.

### [REQUIRED] Step 1: Describe your environment

  * Unity version: 2022.3.8f1
  * Google Mobile Ads Unity plugin version: v8.5.2
  * Platform: iOS
  * Platform OS version: iOS 16.6
  * Any specific devices issue occurs on: x
  * Mediation ad networks used, and their versions: x

#### Steps to reproduce:
Hello, I'm encountering an issue with iOS App Tracking Transparency (ATT).
I recently updated the Google Mobile Ads plugin to version 8.5.2 to implement GDPR Consent.

Here's my application process:
1. Start the app.
2. Initialize Google Mobile Ads Consent.
3. Obtain consent.
4. Request Authorization Tracking.
a. If Authorized: Initialize MobileAds -> Successfully load ads.
b. If Denied: Initialize MobileAds -> **Fail to load ads.**

Before the update, when I was using version 7.1.0,
ads always loaded without any issue related to ATT status.
However, after the update, I'm facing this problem.

I also attempted to load only non-personalized ads by adding the "npa" (Non-Personalized Ads) option to the AdRequest.
Unfortunately, this approach didn't resolve the issue.

Is there something I might be missing?
Please provide assistance and let me know if there are any additional steps I should take.

#### Relevant Code:
    public void LoadAd(string _adUnitId)
    {
        // Clean up the old ad before loading a new one.
        if (_interstitialAd != null)
        {
            DestroyAd();
        }
        Debug.Log("> Loading interstitial ad.");

        // Create our request used to load the ad.
        var adRequest = new AdRequest();
        adRequest.Extras.Add("npa", "1");

        // Send the request to load the ad.
        InterstitialAd.Load(_adUnitId, adRequest, (InterstitialAd ad, LoadAdError error) =>
        {
            // If the operation failed with a reason.
            if (error != null)
            {
                Debug.LogError("> Interstitial ad failed to load an ad with error : " + error);
                return;
            }
            // If the operation failed for unknown reasons.
            // This is an unexpected error, please report this bug if it happens.
            if (ad == null)
            {
                Debug.LogError("> Unexpected error: Interstitial load event fired with null ad and null error.");
                return;
            }

            // The operation completed successfully.
            Debug.Log("> Interstitial ad loaded with response : " + ad.GetResponseInfo());
            _interstitialAd = ad;

            // Register to ad events to extend functionality.
            RegisterEventHandlers(ad);
        });
    }

The rest of the code is from the sample project.

[Error Log]
> Rewarded ad failed to load an ad with error : Error Domain=com.google.admob Code=1 "Request Error: No ad to show." UserInfo={NSLocalizedDescription=Request Error: No ad to show., gad_response_info= ** Response Info **
Response ID: (null)
Network: (null)
** Loaded Adapter Response **
(null)
** Extras Dictionary **
{
}
** Mediation line items **
}
GoogleMobileAds.iOS.RewardedAdClient:RewardedAdFailedToLoadCallback(IntPtr, IntPtr)

> Interstitial ad failed to load an ad with error : Error Domain=com.google.admob Code=1 "Request Error: No ad to show." UserInfo={NSLocalizedDescription=Request Error: No ad to show., gad_response_info= ** Response Info **
Response ID: (null)
Network: (null)
** Loaded Adapter Response **
(null)
** Extras Dictionary **
{
}
** Mediation line items **
}
GoogleMobileAds.iOS.InterstitialClient:InterstitialFailedToLoadCallback(IntPtr, IntPtr)
```
// TODO(you): code here to reproduce the problem
```

**Comment by NVentimiglia** [comment]
Hi @Dooya-log 

This issue looks to be specific to your app. Can you open a support case on the [google group](https://groups.google.com/g/google-admob-ads-sdk) and include a minimal sample, your publisher id, and affected ad unit id(s)?

Thanks.

**Comment by Dooya-log** [comment]
> Hi @Dooya-log
> 
> This issue looks to be specific to your app. Can you open a support case on the [google group](https://groups.google.com/g/google-admob-ads-sdk) and include a minimal sample, your publisher id, and affected ad unit id(s)?
> 
> Thanks.

Thanks for the reply.
As you say, it seems to be a *specific* case because my other apps have no problem under the same conditions.
Even when I tested the same app on different devices, the issue was different.
It will take me a while to find the cause for sure, but if I do, I'll write something further here, thanks for your help :)

**Comment by NVentimiglia** [comment]
Acknowledged,

Looking forward you replication steps.

**Comment by Dooya-log** [comment]
It's been a while! 🖐️
I upgraded the package version to 8.5.3, and finally solved this issue by changing some code.
There seems to be a problem with the ad loading algorithm.
Previously, the next ad was loaded Immediately after show ad like this,

`    
public void ShowAd()
{
    if (_interstitialAd != null && _interstitialAd.CanShowAd())
    {
        Debug.Log("Showing interstitial ad.");
        _interstitialAd.Show();
    }
    else
    {
        Debug.LogError("Interstitial ad is not ready yet.");
    }        
    LoadAd(); // here
}
`

In this flow, it seems that in some cases,
the ad was not displaying properly as the next ad loaded before the ad was served.
So I modified it to load the next ad after the ad is fully completed.

I still don't know why this was affected by ad tracking permissions,
but I'm glad this resolved it. 😅

If anyone is having trouble with similar logic, check this out.
THANKS!
